platform/ios/src/MGLMapView.mm

```diff
@@ -931,6 +856,38 @@ - (void)layoutSubviews
     [self updateUserLocationAnnotationView];
 }
 
+- (void)layoutOrnaments
+{
+    CGFloat x, y;
```

Comment:
This is OK, but just a stylistic tip: `x` and `y` variables that get repurposed multiple times carry the risk that we might fail to update one of these variables and its previous value may “bleed” into a subsequent use of it.

Instead, I sometimes take advantage of the struct initializer syntax to keep the values scoped to where they’re used. For example:

```objc
self.compassView.center = {
    CGRectGetWidth(self.bounds) - CGRectGetMidX(self.compassView.bounds) - self.contentInset.right - 5,
    CGRectGetMidY(self.compassView.bounds) + self.contentInset.top + 5,
};
```

If you’re concerned about accidentally swapping `x` and `y`, you can also use the C99 designated initializer syntax:

```objc
self.compassView.center = {
    .x = CGRectGetWidth(self.bounds) - CGRectGetMidX(self.compassView.bounds) - self.contentInset.right - 5,
    .y = CGRectGetMidY(self.compassView.bounds) + self.contentInset.top + 5,
};
```